Makefile.am

```diff
@@ -44,8 +44,8 @@ AM_CFLAGS =
 if ENABLE_EXTRA_WARNINGS
 AM_CFLAGS += \
 	-Wshadow	\
-	-Wno-initializer-overrides
-
+	-Wno-initializer-overrides \
+	-Wno-error=deprecated-declarations
```

Comment:
A clean compile with Werrors is still required, since the [travis check adds this flag](https://github.com/balabit/syslog-ng/blob/master/.travis.yml#L41).